oauth2client/_helpers.py

```diff
@@ -39,15 +39,42 @@ def _json_encode(data):
   return json.dumps(data, separators=(',', ':'))
 
 
+def _to_bytes(value, encoding='ascii'):
+  """Converts a string value to bytes, if necessary.
+
+  Unfortunately, ``six.b`` is insufficient for this task since in
+  Python2 it does not modify ``unicode`` objects.
+
+  Args:
+      value: The string/bytes value to be converted.
+      encoding: The encoding to use to convert unicode to bytes. Defaults
+                to "ascii", which will not allow any characters from ordinals
+                larger than 127. Other useful values are "latin-1", which
+                which will only allows byte ordinals (up to 255) and "utf-8",
+                which will encode any unicode that needs to be.
+
+  Returns:
+      The original value converted to bytes (if unicode) or as passed in
+      if it started out as bytes.
+
+  Raises:
+      ValueError if the value could not be converted to bytes.
+  """
+  result = value
+  if isinstance(value, six.text_type):
+    result = value.encode(encoding)
+  if not isinstance(result, six.binary_type):
+    raise ValueError('%r could not be converted to bytes' % (value,))
```

Comment:
`str.format` is not properly supported in Python 2.6 (AFAIK). The tupleization is very much intentional, as it avoids spurious errors (e.g. if `value` was already a `tuple`). The fact that `str.__mod__` (e.g. `str %`) accepts a single val instead of a tuple / iterable is kind of a "mistake".